### ISSUE TITLE
How long does it take to render a sample?

### ISSUE THREAD

**Issue by zhuqiangLu** [MMehdiMousavi/SuperCaustics#7]
Hi there,

I am wondering how efficient it is to render a single sample? Can I run it on Ubuntu with a GTX graphic card?

Thank you

**Comment by MMehdiMousavi** [comment]
Hi! great question. SuperCaustics is a real time simulation, so no waiting needed. you can get up to 60 -120 frames per second (or higher depending on your rendering resolution) on an rtx 2060. 

I think you technically could run it with a GTX card, but i imagine it would be very slow, maybe 5 to 10 fps, and definitely no DLSS since GTX cards dont have tensor cores (so you would have to disable the DLSS node) i wouldn't recommend. 

while ive known folks who ran it on linux, i haven't personally tested it, and i strongly suggest you install and run on windows (mainly because the shaders are dx12, so they might not work correctly on Linux.)

**Comment by zhuqiangLu** [comment]
> Hi! great question. SuperCaustics is a real time simulation, so no waiting needed. you can get up to 60 -120 frames per second (or higher depending on your rendering resolution) on an rtx 2060.
> 
> I think you technically could run it with a GTX card, but i imagine it would be very slow, maybe 5 to 10 fps, and definitely no DLSS since GTX cards dont have tensor cores (so you would have to disable the DLSS node) i wouldn't recommend.
> 
> while ive known folks who ran it on linux, i haven't personally tested it, and i strongly suggest you install and run on windows (mainly because the shaders are dx12, so they might not work correctly on Linux.)

Nice! I just managed to get the project running on Linux and it seems all-right at the moment, although it does run in a slow frame rate.

**Comment by zhuqiangLu** [comment]
> Hi! great question. SuperCaustics is a real time simulation, so no waiting needed. you can get up to 60 -120 frames per second (or higher depending on your rendering resolution) on an rtx 2060.
> 
> I think you technically could run it with a GTX card, but i imagine it would be very slow, maybe 5 to 10 fps, and definitely no DLSS since GTX cards dont have tensor cores (so you would have to disable the DLSS node) i wouldn't recommend.
> 
> while ive known folks who ran it on linux, i haven't personally tested it, and i strongly suggest you install and run on windows (mainly because the shaders are dx12, so they might not work correctly on Linux.)

Sorry for reopening this issue, I just ran into another problem. This is my first time to use UE4, so this may be a silly and trivial question. The program crashes every time I hit the play button. I am not sure should I hit the play button for collecting data.

**Comment by zhuqiangLu** [comment]
```
LoginId:61bb72e19bae42c7aed3e3e165c3bdef-000003e8
EpicAccountId:

Fatal error: [File:/home/chenyue/Downloads/UnrealEngine-NvRTX_Caustics-4.26/Engine/Source/Runtime/VulkanRHI/Private/VulkanUtil.cpp] [Line: 803] VulkanRHI::vkQueueSubmit(Queue, 1, &SubmitInfo, Fence->GetHandle()) failed, VkResult=-4 at /home/chenyue/Downloads/UnrealEngine-NvRTX_Caustics-4.26/Engine/Source/Runtime/VulkanRHI/Private/VulkanQueue.cpp:71 with error VK_ERROR_DEVICE_LOST 0x00007f9c4b1336df libUE4Editor-VulkanRHI.so!VulkanRHI::VerifyVulkanResult(VkResult, char const*, char const*, unsigned int) [/home/chenyue/Downloads/UnrealEngine-NvRTX_Caustics-4.26/Engine/Source/Runtime/VulkanRHI/Private/VulkanUtil.cpp:802] 0x00007f9c4b10a816 libUE4Editor-VulkanRHI.so!FVulkanQueue::Submit(FVulkanCmdBuffer*, unsigned int, VkSemaphore_T**) [/home/chenyue/Downloads/UnrealEngine-NvRTX_Caustics-4.26/Engine/Source/Runtime/VulkanRHI/Private/VulkanQueue.cpp:71] 0x00007f9c4b0a132a libUE4Editor-VulkanRHI.so!FVulkanCommandBufferManager::SubmitUploadCmdBuffer(unsigned int, VkSemaphore_T**) [/home/chenyue/Downloads/UnrealEngine-NvRTX_Caustics-4.26/Engine/Source/Runtime/VulkanRHI/Private/VulkanCommandBuffer.cpp:547] 0x00007f9c4b09b1ad libUE4Editor-VulkanRHI.so!FVulkanCommandListContext::RHIBeginTransitions(TArrayView<FRHITransition const*, int>) [/home/chenyue/Downloads/UnrealEngine-NvRTX_Caustics-4.26/Engine/Source/Runtime/VulkanRHI/Private/VulkanBarriers.cpp:923] 0x00007f9d00398dd1 libUE4Editor-RHI.so!FRHICommandResourceTransition::Execute(FRHICommandListBase&) [/home/chenyue/Downloads/UnrealEngine-NvRTX_Caustics-4.26/Engine/Source/Runtime/RHI/Public/RHICommandListCommandExecutes.inl:403] 0x00007f9d029a1480 libUE4Editor-Engine.so!FRHICommand<FRHICommandResourceTransition, FRHICommandResourceTransitionString1514>::ExecuteAndDestruct(FRHICommandListBase&, FRHICommandListDebugContext&) [/home/chenyue/Downloads/UnrealEngine-NvRTX_Caustics-4.26/Engine/Source/Runtime/RHI/Public/RHICommandList.h:768] 0x00007f9d0039b7d0 libUE4Editor-RHI.so!FRHICommandListExecutor::ExecuteInner_DoExecute(FRHICommandListBase&) [/home/chenyue/Downloads/UnrealEngine-NvRTX_Caustics-4.26/Engine/Source/Runtime/RHI/Private/RHICommandList.cpp:373] 0x00007f9d0041b0d8 libUE4Editor-RHI.so!FExecuteRHIThreadTask::DoTask(ENamedThreads::Type, TRefCountPtr<FGraphEvent> const&) [/home/chenyue/Downloads/UnrealEngine-NvRTX_Caustics-4.26/Engine/Source/Runtime/RHI/Private/RHICommandList.cpp:428] 0x00007f9d0041a740 libUE4Editor-RHI.so!TGraphTask<FExecuteRHIThreadTask>::ExecuteTask(TArray<FBaseGraphTask*, TSizedDefaultAllocator<32> >&, ENamedThreads::Type) [/home/chenyue/Downloads/UnrealEngine-NvRTX_Caustics-4.26/Engine/Source/Runtime/Core/Public/Async/TaskGraphInterfaces.h:886] 0x00007f9d05ab016c libUE4Editor-Core.so!FNamedTaskThread::ProcessTasksNamedThread(int, bool) [/home/chenyue/Downloads/UnrealEngine-NvRTX_Caustics-4.26/Engine/Source/Runtime/Core/Private/Async/TaskGraph.cpp:709] 0x00007f9d05aae86e libUE4Editor-Core.so!FNamedTaskThread::ProcessTasksUntilQuit(int) [/home/chenyue/Downloads/UnrealEngine-NvRTX_Caustics-4.26/Engine/Source/Runtime/Core/Private/Async/TaskGraph.cpp:600] 0x00007f9d005d7b8d libUE4Editor-RenderCore.so!FRHIThread::Run() [/home/chenyue/Downloads/UnrealEngine-NvRTX_Caustics-4.26/Engine/Source/Runtime/RenderCore/Private/RenderingThread.cpp:319] 0x00007f9d05b5ff37 libUE4Editor-Core.so!FRunnableThreadPThread::Run() [/home/chenyue/Downloads/UnrealEngine-NvRTX_Caustics-4.26/Engine/Source/Runtime/Core/Private/HAL/PThreadRunnableThread.cpp:25] 0x00007f9d05b24343 libUE4Editor-Core.so!FRunnableThreadPThread::_ThreadProc(void*) [/home/chenyue/Downloads/UnrealEngine-NvRTX_Caustics-4.26/Engine/Source/Runtime/Core/Private/HAL/PThreadRunnableThread.h:185] 0x00007f9d064a56db libpthread.so.0!UnknownFunction(0x76da) 0x00007f9cfcc3f61f libc.so.6!clone(+0x3e)

libUE4Editor-Core.so!FGenericPlatformMisc::RaiseException(unsigned int) [/home/chenyue/Downloads/UnrealEngine-NvRTX_Caustics-4.26/Engine/Source/Runtime/Core/Private/GenericPlatform/GenericPlatformMisc.cpp:472]
libUE4Editor-Core.so!FOutputDevice::LogfImpl(char16_t const*, ...) [/home/chenyue/Downloads/UnrealEngine-NvRTX_Caustics-4.26/Engine/Source/Runtime/Core/Private/Misc/OutputDevice.cpp:61]
libUE4Editor-VulkanRHI.so!VulkanRHI::VerifyVulkanResult(VkResult, char const*, char const*, unsigned int) [/home/chenyue/Downloads/UnrealEngine-NvRTX_Caustics-4.26/Engine/Source/Runtime/VulkanRHI/Private/VulkanUtil.cpp:802]
libUE4Editor-VulkanRHI.so!FVulkanQueue::Submit(FVulkanCmdBuffer*, unsigned int, VkSemaphore_T**) [/home/chenyue/Downloads/UnrealEngine-NvRTX_Caustics-4.26/Engine/Source/Runtime/VulkanRHI/Private/VulkanQueue.cpp:71]
libUE4Editor-VulkanRHI.so!FVulkanCommandBufferManager::SubmitUploadCmdBuffer(unsigned int, VkSemaphore_T**) [/home/chenyue/Downloads/UnrealEngine-NvRTX_Caustics-4.26/Engine/Source/Runtime/VulkanRHI/Private/VulkanCommandBuffer.cpp:547]
libUE4Editor-VulkanRHI.so!FVulkanCommandListContext::RHIBeginTransitions(TArrayView<FRHITransition const*, int>) [/home/chenyue/Downloads/UnrealEngine-NvRTX_Caustics-4.26/Engine/Source/Runtime/VulkanRHI/Private/VulkanBarriers.cpp:923]
libUE4Editor-RHI.so!FRHICommandResourceTransition::Execute(FRHICommandListBase&) [/home/chenyue/Downloads/UnrealEngine-NvRTX_Caustics-4.26/Engine/Source/Runtime/RHI/Public/RHICommandListCommandExecutes.inl:403]
libUE4Editor-Engine.so!FRHICommand<FRHICommandResourceTransition, FRHICommandResourceTransitionString1514>::ExecuteAndDestruct(FRHICommandListBase&, FRHICommandListDebugContext&) [/home/chenyue/Downloads/UnrealEngine-NvRTX_Caustics-4.26/Engine/Source/Runtime/RHI/Public/RHICommandList.h:768]
libUE4Editor-RHI.so!FRHICommandListExecutor::ExecuteInner_DoExecute(FRHICommandListBase&) [/home/chenyue/Downloads/UnrealEngine-NvRTX_Caustics-4.26/Engine/Source/Runtime/RHI/Private/RHICommandList.cpp:373]
libUE4Editor-RHI.so!FExecuteRHIThreadTask::DoTask(ENamedThreads::Type, TRefCountPtr<FGraphEvent> const&) [/home/chenyue/Downloads/UnrealEngine-NvRTX_Caustics-4.26/Engine/Source/Runtime/RHI/Private/RHICommandList.cpp:428]
libUE4Editor-RHI.so!TGraphTask<FExecuteRHIThreadTask>::ExecuteTask(TArray<FBaseGraphTask*, TSizedDefaultAllocator<32> >&, ENamedThreads::Type) [/home/chenyue/Downloads/UnrealEngine-NvRTX_Caustics-4.26/Engine/Source/Runtime/Core/Public/Async/TaskGraphInterfaces.h:886]
libUE4Editor-Core.so!FNamedTaskThread::ProcessTasksNamedThread(int, bool) [/home/chenyue/Downloads/UnrealEngine-NvRTX_Caustics-4.26/Engine/Source/Runtime/Core/Private/Async/TaskGraph.cpp:709]
libUE4Editor-Core.so!FNamedTaskThread::ProcessTasksUntilQuit(int) [/home/chenyue/Downloads/UnrealEngine-NvRTX_Caustics-4.26/Engine/Source/Runtime/Core/Private/Async/TaskGraph.cpp:600]
libUE4Editor-RenderCore.so!FRHIThread::Run() [/home/chenyue/Downloads/UnrealEngine-NvRTX_Caustics-4.26/Engine/Source/Runtime/RenderCore/Private/RenderingThread.cpp:319]
libUE4Editor-Core.so!FRunnableThreadPThread::Run() [/home/chenyue/Downloads/UnrealEngine-NvRTX_Caustics-4.26/Engine/Source/Runtime/Core/Private/HAL/PThreadRunnableThread.cpp:25]
libUE4Editor-Core.so!FRunnableThreadPThread::_ThreadProc(void*) [/home/chenyue/Downloads/UnrealEngine-NvRTX_Caustics-4.26/Engine/Source/Runtime/Core/Private/HAL/PThreadRunnableThread.h:185]
libpthread.so.0!UnknownFunction(0x76da)
libc.so.6!clone(+0x3e)
```

Here is the error message, unfortunately I don't have a Windows machine to test the project, so I am not sure if this is caused by the Ubuntu or something else.

**Comment by zhuqiangLu** [comment]
It looks like this error has something to do with ray tracing (check [Here](https://stackoverflow.com/questions/65865327/trouble-in-launching-unreal-editor-4-built-on-ubuntu-20-04)). 
Disabling ray tracing gives me another error 
```
LoginId:61bb72e19bae42c7aed3e3e165c3bdef-000003e8
EpicAccountId:

Assertion failed: bSuccess [File:/home/chenyue/Downloads/UnrealEngine-NvRTX_Caustics-4.26/Engine/Source/Runtime/VulkanRHI/Private/VulkanCommandBuffer.cpp] [Line: 458]

libUE4Editor-Core.so!FGenericPlatformMisc::RaiseException(unsigned int) [/home/chenyue/Downloads/UnrealEngine-NvRTX_Caustics-4.26/Engine/Source/Runtime/Core/Private/GenericPlatform/GenericPlatformMisc.cpp:472]
libUE4Editor-Core.so!FOutputDevice::LogfImpl(char16_t const*, ...) [/home/chenyue/Downloads/UnrealEngine-NvRTX_Caustics-4.26/Engine/Source/Runtime/Core/Private/Misc/OutputDevice.cpp:61]
libUE4Editor-Core.so!AssertFailedImplV(char const*, char const*, int, char16_t const*, __va_list_tag*) [/home/chenyue/Downloads/UnrealEngine-NvRTX_Caustics-4.26/Engine/Source/Runtime/Core/Private/Misc/AssertionMacros.cpp:104]
libUE4Editor-Core.so!FDebug::CheckVerifyFailedImpl(char const*, char const*, int, char16_t const*, ...) [/home/chenyue/Downloads/UnrealEngine-NvRTX_Caustics-4.26/Engine/Source/Runtime/Core/Private/Misc/AssertionMacros.cpp:458]
libUE4Editor-VulkanRHI.so!FVulkanCommandBufferManager::WaitForCmdBuffer(FVulkanCmdBuffer*, float) [/home/chenyue/Downloads/UnrealEngine-NvRTX_Caustics-4.26/Engine/Source/Runtime/VulkanRHI/Private/VulkanCommandBuffer.cpp:458]
libUE4Editor-VulkanRHI.so!FVulkanDynamicRHI::RHIReadSurfaceFloatData(FRHITexture*, FIntRect, TArray<FFloat16Color, TSizedDefaultAllocator<32> >&, ECubeFace, int, int) [/home/chenyue/Downloads/UnrealEngine-NvRTX_Caustics-4.26/Engine/Source/Runtime/VulkanRHI/Private/VulkanRenderTarget.cpp:597]
libUE4Editor-RHI.so!FDynamicRHI::RHIReadSurfaceFloatData_RenderThread(FRHICommandListImmediate&, FRHITexture*, FIntRect, TArray<FFloat16Color, TSizedDefaultAllocator<32> >&, ECubeFace, int, int) [/home/chenyue/Downloads/UnrealEngine-NvRTX_Caustics-4.26/Engine/Source/Runtime/RHI/Private/RHICommandList.cpp:2700]
libUE4Editor-Renderer.so!ComputeSingleAverageBrightnessFromCubemap(FRHICommandListImmediate&, ERHIFeatureLevel::Type, int, FSceneRenderTargetItem&) [/home/chenyue/Downloads/UnrealEngine-NvRTX_Caustics-4.26/Engine/Source/Runtime/Renderer/Private/ReflectionEnvironmentCapture.cpp:328]
libUE4Editor-Renderer.so!ComputeAverageBrightness(FRHICommandListImmediate&, ERHIFeatureLevel::Type, int, float&) [/home/chenyue/Downloads/UnrealEngine-NvRTX_Caustics-4.26/Engine/Source/Runtime/Renderer/Private/ReflectionEnvironmentCapture.cpp:348]
libUE4Editor-Renderer.so!TEnqueueUniqueRenderCommandType<FScene::UpdateSkyCaptureContents(USkyLightComponent const*, bool, UTextureCube*, FTexture*, float&, TSHVectorRGB<3>&, TArray<FFloat16Color, TSizedDefaultAllocator<32> >*)::FilterCommandName, FScene::UpdateSkyCaptureContents(USkyLightComponent const*, bool, UTextureCube*, FTexture*, float&, TSHVectorRGB<3>&, TArray<FFloat16Color, TSizedDefaultAllocator<32> >*)::$_24>::DoTask(ENamedThreads::Type, TRefCountPtr<FGraphEvent> const&) [/home/chenyue/Downloads/UnrealEngine-NvRTX_Caustics-4.26/Engine/Source/Runtime/RenderCore/Public/RenderingThread.h:183]
libUE4Editor-Renderer.so!TGraphTask<TEnqueueUniqueRenderCommandType<FScene::UpdateSkyCaptureContents(USkyLightComponent const*, bool, UTextureCube*, FTexture*, float&, TSHVectorRGB<3>&, TArray<FFloat16Color, TSizedDefaultAllocator<32> >*)::FilterCommandName, FScene::UpdateSkyCaptureContents(USkyLightComponent const*, bool, UTextureCube*, FTexture*, float&, TSHVectorRGB<3>&, TArray<FFloat16Color, TSizedDefaultAllocator<32> >*)::$_24> >::ExecuteTask(TArray<FBaseGraphTask*, TSizedDefaultAllocator<32> >&, ENamedThreads::Type) [/home/chenyue/Downloads/UnrealEngine-NvRTX_Caustics-4.26/Engine/Source/Runtime/Core/Public/Async/TaskGraphInterfaces.h:886]
libUE4Editor-Core.so!FNamedTaskThread::ProcessTasksNamedThread(int, bool) [/home/chenyue/Downloads/UnrealEngine-NvRTX_Caustics-4.26/Engine/Source/Runtime/Core/Private/Async/TaskGraph.cpp:709]
libUE4Editor-Core.so!FNamedTaskThread::ProcessTasksUntilQuit(int) [/home/chenyue/Downloads/UnrealEngine-NvRTX_Caustics-4.26/Engine/Source/Runtime/Core/Private/Async/TaskGraph.cpp:600]
libUE4Editor-RenderCore.so!RenderingThreadMain(FEvent*) [/home/chenyue/Downloads/UnrealEngine-NvRTX_Caustics-4.26/Engine/Source/Runtime/RenderCore/Private/RenderingThread.cpp:372]
libUE4Editor-RenderCore.so!FRenderingThread::Run() [/home/chenyue/Downloads/UnrealEngine-NvRTX_Caustics-4.26/Engine/Source/Runtime/RenderCore/Private/RenderingThread.cpp:526]
libUE4Editor-Core.so!FRunnableThreadPThread::Run() [/home/chenyue/Downloads/UnrealEngine-NvRTX_Caustics-4.26/Engine/Source/Runtime/Core/Private/HAL/PThreadRunnableThread.cpp:25]
libUE4Editor-Core.so!FRunnableThreadPThread::_ThreadProc(void*) [/home/chenyue/Downloads/UnrealEngine-NvRTX_Caustics-4.26/Engine/Source/Runtime/Core/Private/HAL/PThreadRunnableThread.h:185]
libpthread.so.0!UnknownFunction(0x76da)
libc.so.6!clone(+0x3e)


```

**Comment by MMehdiMousavi** [comment]
Hi, this looks like a Vulkan problem, Also try disabling the DLSS node as well, could be causing a crash since GTX cards don't have tensor cores. 
But ultimately, I suggest running things on Windows instead. there's just better support for ue4 on windows.

**Comment by zhuqiangLu** [comment]
> Hi, this looks like a Vulkan problem, Also try disabling the DLSS node as well, could be causing a crash since GTX cards don't have tensor cores. But ultimately, I suggest running things on Windows instead. there's just better support for ue4 on windows.

Thank you for your advise, the windows machine with a RTX graphic card is on its way now. It looks like a DX12 related problem, I tried this project on a windows machine and it runs just fine. 

Is it normal to have a long export time for a scene? I can render the scene in UE4 editor, but it takes quite some time to export it(So I canceled it). Can I run Probe.py with the preview render scene?
